src/Download/Downloader.cpp

```diff
@@ -103,50 +136,49 @@ namespace {
             }
         }
 
         long response_code;
         FILE* fp = fopen(FILE_PATH, "wb");
         if(!fp) {
             sua::Logger::critical("Failed to open '{}' for writing", FILE_PATH);
-            return sua::TechCode::DownloadFailed;
+            return std::make_tuple(sua::TechCode::DownloadFailed, "temporary file could not be opened for writing");
         }
 
-        curl_easy_setopt(easy_handle, CURLOPT_URL, url);
-        curl_easy_getinfo(easy_handle, CURLINFO_RESPONSE_CODE, &response_code);
+        curl_easy_setopt(h, CURLOPT_URL, url);
+        curl_easy_getinfo(h, CURLINFO_RESPONSE_CODE, &response_code);
         if(!caFile.empty()) {
-            curl_easy_setopt(easy_handle, CURLOPT_CAINFO, caFile.c_str());
+            curl_easy_setopt(h, CURLOPT_CAINFO, caFile.c_str());
         } else {
-            curl_easy_setopt(easy_handle, CURLOPT_CAPATH, caPath.c_str());
+            curl_easy_setopt(h, CURLOPT_CAPATH, caPath.c_str());
         }
-        curl_easy_setopt(easy_handle, CURLOPT_SSL_VERIFYPEER, 1L);
-        curl_easy_setopt(easy_handle, CURLOPT_WRITEFUNCTION, write_data);
-        curl_easy_setopt(easy_handle, CURLOPT_WRITEDATA, fp);
-        curl_easy_setopt(easy_handle, CURLOPT_FOLLOWLOCATION, 1L);
-        curl_easy_setopt(easy_handle, CURLOPT_PROGRESSDATA, &data);
-        curl_easy_setopt(easy_handle, CURLOPT_PROGRESSFUNCTION, progress_callback);
-        curl_easy_setopt(easy_handle, CURLOPT_NOPROGRESS, 0);
-        curl_easy_setopt(easy_handle, CURLOPT_PROTOCOLS_STR, "https");
-        CURLcode res = curl_easy_perform(easy_handle);
+        curl_easy_setopt(h, CURLOPT_SSL_VERIFYPEER, 1L);
+        curl_easy_setopt(h, CURLOPT_WRITEFUNCTION, write_data);
+        curl_easy_setopt(h, CURLOPT_WRITEDATA, fp);
+        curl_easy_setopt(h, CURLOPT_FOLLOWLOCATION, 1L);
+        curl_easy_setopt(h, CURLOPT_PROGRESSDATA, &data);
+        curl_easy_setopt(h, CURLOPT_PROGRESSFUNCTION, progress_callback);
+        curl_easy_setopt(h, CURLOPT_NOPROGRESS, 0);
+        curl_easy_setopt(h, CURLOPT_PROTOCOLS_STR, "https");
+        CURLcode res = curl_easy_perform(h);
 
         sua::Logger::debug("curl_easy_perform ended with code = '{}'", res);
 
         long http_code = 0;
-        curl_easy_getinfo(easy_handle, CURLINFO_RESPONSE_CODE, &http_code);
-        curl_easy_cleanup(easy_handle);
-        curl_global_cleanup();
+        curl_easy_getinfo(h, CURLINFO_RESPONSE_CODE, &http_code);
         fclose(fp);
         progressNotificationLimiter = 0;
 
         sua::Logger::debug("CURLINFO_RESPONSE_CODE = {}", http_code);
         if(http_code != 200) {
-            sua::Logger::error(curl_easy_strerror(res));
-            return sua::TechCode::DownloadFailed;
+            auto e = curl_easy_strerror(res);
+            sua::Logger::error(e);
+            return std::make_tuple(sua::TechCode::DownloadFailed, "");
```

Comment:
would add "Errorcode = ...", to get later not an error message with text "Download failed:"

Reply:
Added error to 2nd value in tuple